### PR TITLE
chore(ci): auto doc labeler job

### PR DESCRIPTION
**Pull request by waynexia**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Add an action job to auto label "Doc not needed" on PR based on the checkbox in "Checklist".

This patch also adds a dedicated checkbox in PR template for "doc not needed"/"doc update required" tuples.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
Closes https://github.com/GreptimeTeam/greptimedb/issues/2997

### CODE REVIEW

File: .github/doc-label-config.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Doc not needed:
+    - '- \[x\]  I have written the necessary rustdoc comments'
```
Comment (waynexia) [inline .github/doc-label-config.yml:2]: I use this for now, but it's not very accurate. Maybe we can add the third checkbox for this usage.

**Review by tisonkun — COMMENTED** [review]
+1 for giving it a try.

**Comment by waynexia** [comment]
I've verified this config on https://github.com/waynexia/greptimedb/pull/1

**Review by sunng87 — APPROVED** [review]

**Comment by sunng87** [comment]
Great! Never thought it has an official action for this.

**Comment by waynexia** [comment]
> Great! Never thought it has an official action for this.

Me too, I think pulsar also needs this :yum:

**Review by nicecui — APPROVED** [review]
Wow! Great job! 👍

**Pull request merged**